pkg/controller/elasticsearch/client/base.go

```diff
@@ -108,12 +108,59 @@ func (c *baseClient) request(
 		body = bytes.NewBuffer(outData)
 	}
 
+	request, err := http.NewRequest(method, stringsutil.Concat(c.Endpoint, pathWithQuery), body)
+	if err != nil {
+		return err
+	}
+
+	resp, err := c.doRequest(ctx, request)
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+
+	if responseObj != nil {
+		if err := json.NewDecoder(resp.Body).Decode(responseObj); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// unsafeRequest performs a new http request and decodes the payload if the status code of the response is 2xx or if the
+// given skipErrFunc function returns true.
+//
+// if requestObj is not nil, it's marshalled as JSON and used as the request body
+// if responseObj is not nil, it should be a pointer to an struct. The response body will be unmarshalled from JSON
+// into this struct if the status code of the response is 2xx or if the given skipErrFunc function returns true.
+func (c *baseClient) unsafeRequest(
```

Comment:
Can we find another name for this? Is it really unsafe to parse the body in certain error cases? Idk maybe `requestIgnoringError(context.Context, func(error) bool, string, interface{}, interface{})` 

An alternative idea would be to modify the existing request function to take a predicate that by default only returns false. But I guess that would be a bit more intrusive as you would have to change all invocations.

Reply:
> Can we find another name for this? Is it really unsafe to parse the body in certain error cases?

Sure but I don't find a good name! I think it could be unsafe depending the `checkSkipErr`. With this name I wanted to highlight that you really need know what you are doing if you use this function.

> An alternative idea would be to modify the existing request function to take a predicate that by default only returns 
false. But I guess that would be a bit more intrusive as you would have to change all invocations.

That was my initial thought but looking at it again, given that we have little intermediates functions get/put/post/delete functions, it is not that intrusive. I'll update the PR in that direction.